corehq/messaging/scheduling/models/timed_schedule.py

```diff
@@ -0,0 +1,197 @@
+import calendar
+from corehq.messaging.scheduling.exceptions import InvalidMonthlyScheduleConfiguration
+from corehq.messaging.scheduling.models.abstract import Schedule, Event, Broadcast
+from corehq.util.timezones.conversions import ServerTime, UserTime
+from datetime import timedelta, datetime, date
+from dimagi.utils.decorators.memoized import memoized
+from django.db import models, transaction
+
+
+class TimedSchedule(Schedule):
+    REPEAT_INDEFINITELY = -1
+    MONTHLY = -1
+
+    schedule_length = models.IntegerField()
+    total_iterations = models.IntegerField()
+
+    @property
+    @memoized
+    def memoized_events(self):
+        """
+        This is named with a memoized_ prefix to be clear that it should only be used
+        when the event set is not changing.
+        """
+        return list(self.timedevent_set.order_by('order'))
+
+    def set_first_event_due_timestamp(self, instance, start_date=None):
+        """
+        If start_date is None, we set it automatically ensuring that
+        self.next_event_due does not get set in the past for the first
+        event.
+        """
+        if start_date:
+            instance.start_date = start_date
+        else:
+            instance.start_date = ServerTime(datetime.utcnow()).user_time(instance.timezone).done().date()
```

Comment:
What's the purpose of this conversion? I would have assumed that the dates would all be stored in UTC.